src/polytopes/simplex.jl

```diff
@@ -37,3 +37,16 @@ nvertices(::Type{<:Simplex{K}}) where {K} = K + 1
 
 Base.isapprox(s₁::Simplex, s₂::Simplex; kwargs...) =
   all(isapprox(v₁, v₂; kwargs...) for (v₁, v₂) in zip(vertices(s₁), vertices(s₂)))
+
+function normal(splx::Simplex{K,Dim,T}) where {K,Dim,T<:Real}
+  Dim > K || throw(
+    ArgumentError("simplex embedding dimension must be larger than parametric dimension for a normal vector to exist")
+  )
+  # It turns out the QR decomposition can be used to find an orthogonal basis,
+  # where by construction the nth column of `Q` is orthogonal to columns 1:(n-1).
+  # Further, `qr` guarantees `Q` to be unitary, therefore `norm(Q[:, i]) ≈ 1` for all columns.
+  p1, pothers... = vertices(splx)
+  basis = [(p - p1 for p in pothers)... rand!(similar(coordinates(p1)))]
```

Comment:
There is a small chance that this random vector will live in the subspace spanned by the first vectors, right?Should we add a `while` loop until linear independence is met?

Reply:
> Should we add a while loop until linear independence is met?

The qr decomposition will work even if the last vector is in the subspace (basically the `while` condition is baked into the `qr` function.) Proof:

```julia
using LinearAlgebra
e1 = [1;0;0]
qr([e1 e1 e1]).Q |> Matrix

#3×3 Matrix{Float64}:
# 1.0  0.0  0.0
# 0.0  1.0  0.0
# 0.0  0.0  1.0

```

Reply:
Replaced `splat` with `stack`. Just list comprehension doesn't really work.

Benchmarks show 10x speedup over splatted version when dimensionality is 100, but the runtime is approximately the same for dimensionality 3. I think this is premature optimization (since beyond dimension 10 we run into a bunch of problems anyways), but it also doesn't hurt.